java/src/shared/model/bank/ResourceCardBank.java

```diff
@@ -52,14 +88,264 @@ public ResourceCardBank (Player player) {
      * @param cardToAdd ResourceCard to add
      */
     public void addResource(ResourceCard cardToAdd) {
-        resourceCards.add(cardToAdd);
+        switch (cardToAdd.getType()) {
+            case WOOD:
+                woods.add((Wood) cardToAdd);
+                break;
+            case BRICK:
+                bricks.add((Brick) cardToAdd);
+                break;
+            case SHEEP:
+                sheeps.add((Sheep) cardToAdd);
+                break;
+            case WHEAT:
+                wheats.add((Wheat) cardToAdd);
+                break;
+            case ORE:
+                ores.add((Ore) cardToAdd);
+                break;
+        }
+    }
+
+    @Override
+    public ResourceCard draw(ResourceType type) throws InvalidTypeException, Exception {
+        if (ownedByGame) {
+            switch (type) {
+                case BRICK:
+                    if (bricks.size() > 0) {
+                        return bricks.remove(0);
+                    } else {
+                        return null;
```

Comment:
Perhaps we should throw a `NoResourcesAvailableException` or something similar instead of returning `null`?